tests/filecheck/dialects/riscv/riscv_assembly_emission.mlir

```diff
@@ -1,178 +1,187 @@
+    // CHECK-NEXT: slti j1, j1, 1
+    %sltiu = "riscv.sltiu"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+    // CHECK-NEXT: sltiu j1, j1, 1
+    %andi = "riscv.andi"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+    // CHECK-NEXT: andi j1, j1, 1
+    %ori = "riscv.ori"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+    // CHECK-NEXT: ori j1, j1, 1
+    %xori = "riscv.xori"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+    // CHECK-NEXT: xori j1, j1, 1
+    %slli = "riscv.slli"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+    // CHECK-NEXT: slli j1, j1, 1
+    %srli = "riscv.srli"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+    // CHECK-NEXT: srli j1, j1, 1
+    %srai = "riscv.srai"(%1) {"immediate" = 1 : i32}: (!riscv.reg<j1>) -> !riscv.reg<j1>
+    // CHECK-NEXT: srai j1, j1, 1
+    %lui = "riscv.lui"() {"immediate" = 1 : i32}: () -> !riscv.reg<j0>
+    // CHECK-NEXT: lui j0, 1
+    %auipc = "riscv.auipc"() {"immediate" = 1 : i32}: () -> !riscv.reg<j0>
+    // CHECK-NEXT: auipc j0, 1
+
+    // Integer Register-Register Operations
+    %add = "riscv.add"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: add j2, j2, j1
+    %slt = "riscv.slt"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: slt j2, j2, j1
+    %sltu = "riscv.sltu"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sltu j2, j2, j1
+    %and = "riscv.and"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: and j2, j2, j1
+    %or = "riscv.or"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: or j2, j2, j1
+    %xor = "riscv.xor"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: xor j2, j2, j1
+    %sll = "riscv.sll"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sll j2, j2, j1
+    %srl = "riscv.srl"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: srl j2, j2, j1
+    %sub = "riscv.sub"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sub j2, j2, j1
+    %sra = "riscv.sra"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sra j2, j2, j1
+    "riscv.nop"() : () -> ()
+    // CHECK-NEXT: nop
+
+    // RV32I/RV64I: 2.5 Control Transfer Instructions
+
+    // Unconditional Branch Instructions
+    "riscv.jal"() {"immediate" = 1 : i32} : () -> ()
+    // CHECK-NEXT: jal 1
+    "riscv.jal"() {"immediate" = 1 : i32, "rd" = !riscv.reg<s0>} : () -> ()
+    // CHECK-NEXT: jal s0, 1
+    "riscv.jal"() {"immediate" = #riscv.label<"label">} : () -> ()
+    // CHECK-NEXT: jal label
+
+    "riscv.j"() {"immediate" = 1 : i32, "rd" = !riscv.reg<zero>} : () -> ()
+    // CHECK-NEXT: j 1
+    "riscv.j"() {"immediate" = #riscv.label<"label">, "rd" = !riscv.reg<zero>} : () -> ()
```

Comment:
Shouldn't this then be a block not a label?

Reply:
that's also in the next PR